packages/server/src/sdk/app/views/index.ts

```diff
@@ -40,16 +40,87 @@ export async function getEnriched(viewId: string): Promise<ViewV2Enriched> {
   return pickApi(tableId).getEnriched(viewId)
 }
 
+export async function getTable(view: string | ViewV2): Promise<Table> {
+  const viewId = typeof view === "string" ? view : view.id
+  const cached = context.getTableForView(viewId)
+  if (cached) {
+    return cached
+  }
+  const { tableId } = utils.extractViewInfoFromID(viewId)
+  const table = await sdk.tables.getTable(tableId)
+  context.setTableForView(viewId, table)
+  return table
+}
+
+export function isView(view: any): view is ViewV2 {
+  return (
+    view.version === 2 && "id" in view && "tableId" in view && "name" in view
+  )
+}
+
+async function guardCalculationViewSchema(
+  table: Table,
+  view: Omit<ViewV2, "id" | "version">
+) {
+  const calculationFields = helpers.views.calculationFields(view)
+  for (const calculationFieldName of Object.keys(calculationFields)) {
+    const schema = calculationFields[calculationFieldName]
+    const targetSchema = table.schema[schema.field]
+    if (!targetSchema) {
+      throw new HTTPError(
+        `Calculation field "${calculationFieldName}" references field "${schema.field}" which does not exist in the table schema`,
+        400
+      )
+    }
+
+    if (!helpers.schema.isNumeric(targetSchema)) {
+      throw new HTTPError(
+        `Calculation field "${calculationFieldName}" references field "${schema.field}" which is not a numeric field`,
+        400
+      )
+    }
+  }
+
+  const groupByFields = helpers.views.basicFields(view)
+  for (const groupByFieldName of Object.keys(groupByFields)) {
+    const targetSchema = table.schema[groupByFieldName]
+    if (!targetSchema) {
+      throw new HTTPError(
+        `Group by field "${groupByFieldName}" does not exist in the table schema`,
```

Comment:
should this error message state that it doesn't exist or is an invalid type, since the `basicFields` are the only ones returned, it could be a relationship column etc

Reply:
I'm not sure I understand what you mean. The current wording is strictly correct when compared against what the code is doing. I'm uncertain how relationship fields come into play.

Reply:
Ah I think I mis-understood this functions purpose - is it for validating the view when its being saved? I was basing this on what I thought the `basicFields` function does, but I now realise this simply retrieves all the fields which aren't calculation type from the schema - I understand now - sorry!

Reply:
Actually - raises a separate point, the group-by and calculation columns can come from a related table, so we need to validate the columns exist in the main table, or a table they are related to (making sure they correctly reference that other table).

Reply:
Yeah this is for when views get saved. `basicFields` maybe not the best name. 😅